.github/workflows/pythonpackage.yml

```diff
@@ -43,3 +43,9 @@ jobs:
         run: |
           source $VENV
           python e2e_tests/sandbox_basic_test.py basic 2.0
+      - name: Upload snapshot report
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: snapshot-report-textual
+          path: tests/snapshot_tests/output/snapshot_report.html
```

Comment:
This runs on both failure and success. On success, there's no snapshot report, but the step still succeeds.